app/org/sagebionetworks/bridge/validators/ActivityValidator.java

```diff
@@ -0,0 +1,62 @@
+package org.sagebionetworks.bridge.validators;
+
+import static org.apache.commons.lang3.StringUtils.isBlank;
+
+import org.sagebionetworks.bridge.models.schedules.Activity;
+import org.sagebionetworks.bridge.models.schedules.ActivityType;
+import org.sagebionetworks.bridge.models.schedules.SurveyReference;
+import org.springframework.validation.Errors;
+import org.springframework.validation.Validator;
+
+public class ActivityValidator implements Validator {
+    
+    private static final String CANNOT_BE_BLANK = "cannot be missing, null, or blank";
+    private static final String CANNOT_BE_NULL = "cannot be null";
+    private static final String ACTIVITY_TYPE_PROPERTY = "activityType";
+    private static final String REF_PROPERTY = "ref";
+    private static final String SURVEY_PROPERTY = "survey";
+    private static final String GUID_PROPERTY = "guid";
+    private static final String CREATED_ON_PROPERTY = "createdOn";
+    
+    @Override
+    public boolean supports(Class<?> clazz) {
+        return Activity.class.isAssignableFrom(clazz);
+    }
+
+    @Override
+    public void validate(Object obj, Errors errors) {
+        Activity activity = (Activity)obj;
+        if (activity.getActivityType() == null) {
+            errors.rejectValue(ACTIVITY_TYPE_PROPERTY, CANNOT_BE_NULL);
+        }
+        if (isBlank(activity.getLabel())) {
+            errors.rejectValue("label", CANNOT_BE_BLANK);
+        }
```

Comment:
This was just moving activity validation out of the schedule validator, because it was getting complicated. I think it is worth testing for empty string labels... if someone does that through a UI, they are going to get a UI that's unusable, because the label is likely going to be used as the basis for some affordance.